### PR TITLE
Assessment overview bonus xp

### PR DESCRIPTION
**Pull request by DesSnowy**
**Description**
This pull request supports the front end PR which fixes a bug where no bonus XP assignments would display additional XP messages upon finalisation. This was done by adding earlySubmissionXp when render Overview.json. 

**Changed made**
- [X] Added earlySubmissionXp to Overview
- [X] Changed test cases accordingly

### CODE REVIEW

**Comment by DesSnowy** [comment]
Corresponding frontend PR: https://github.com/source-academy/frontend/pull/2821

**Comment by RichDom2185** [comment]
Hmm, could you investigate why merging `master` somehow resulted in failing tests?

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks! We wait for the FE to pass review before merging both at the same time.

**Pull request merged**
